### PR TITLE
fix(Toolbar): allow popper props in the overflow item

### LINKED ISSUE #14721: Toolbar’s overflowItem does not apply positioning props when provided
### Environment Information

- **Package version(s)**: `@fluentui/react-northstar@0.51.0`
- **Browser and OS versions**: all

### Please provide a reproduction of the bug in a codepen:

https://codesandbox.io/s/32ho2

#### Actual behavior:

Typechecks for `menu: { offset: [0, 4], position: 'below' }` in the `overflowItem` prop pass, but the menu will still display above the invoker and offset does not appear to be applied.

#### Expected behavior:

The `overflowItem`'s menu should apply the positioning props provided: e.g. when `position: 'below'` is provided, the overflow menu should always be positioned below the invoker.

### Priorities and help requested:

Are you willing to submit a PR to fix? Yes, but I would need some support from an engineer more familiar with this codebase.

Requested priority: Normal

Products/sites affected: Teams

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #14721
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR allows to set the `popper` properties for `menu` in the `OverflowItem`. 

Usage:

```jsx
<Toolbar
      aria-label="Toolbar overflow menu"
      items={toolbarItems}
      overflow
      overflowOpen={overflowOpen}
      overflowItem={{ title: 'More', menu: { popper: { offset: [0, 4], position: 'below' }} }}
      onOverflowOpenChange={(e, { overflowOpen }) => {
        setOverflowOpen(overflowOpen)
      }}
      getOverflowItems={startIndex => itemData.slice(startIndex)}
    />

```

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx [hunk 1]
```diff
@@ -62,4 +62,24 @@ describe('Toolbar', () => {
...
   });
+
+  describe('Overflow Toolbar', () => {
+    it('Should override popper properties', () => {
+      const wrapper = mountWithProvider(
+        <Toolbar
+          aria-label="Toolbar overflow menu"
+          items={[]}
+          overflow
+          overflowItem={{ title: 'More', menu: { popper: { offset: [0, 4], position: 'below' } } }}
```
Comment (ling1726) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:73]: nit: you can create the `menu` object at the beginning of the test, so you don't have to repeat yourself

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx [hunk 2]
```diff
@@ -62,4 +62,24 @@ describe('Toolbar', () => {
       expect(getRenderedAttribute(italicToolbarButton, 'aria-pressed', '')).toBe('false');
     });
   });
+
+  describe('Overflow Toolbar', () => {
```
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:66]:
```suggestion
  describe('overflowItem', () => {
```

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx [hunk 3]
```diff
@@ -62,4 +62,24 @@ describe('Toolbar', () => {
...
+    it('Should override popper properties', () => {
```
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:67]:
```suggestion
    it('popper props can be overriden', () => {
```

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx [hunk 4]
```diff
@@ -62,4 +62,24 @@ describe('Toolbar', () => {
...
+      const wrapper = mountWithProvider(
+        <Toolbar
+          aria-label="Toolbar overflow menu"
```
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:70]:
```suggestion
```

It's not required for tests

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx (hunk 1 above)
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:73]:
```suggestion
          overflowItem={{ menu: { popper: { offset: [0, 4], position: 'below' } } }}
```

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx [hunk 5]
```diff
@@ -62,4 +62,24 @@ describe('Toolbar', () => {
...
+          items={[]}
+          overflow
```
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:72]:
```suggestion
          overflow
          overflowOpen
```

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx [hunk 6]
```diff
@@ -62,4 +62,24 @@ describe('Toolbar', () => {
...
+          overflowItem={{ title: 'More', menu: { popper: { offset: [0, 4], position: 'below' } } }}
+        />,
+      );
+
+      const overflowItem = wrapper.find('ToolbarItem').last();
+
+      expect(overflowItem.prop('menu')).toMatchObject({
+        items: [],
+        popper: { positionFixed: true, offset: [0, 4], position: 'below' }, // it should have default positionFixed prop and the new props passed
+      });
```
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:82]:
```suggestion
      expect(wrapper.find('Popper').prop('positionFixed').toBe(true)
      expect(wrapper.find('Popper').prop('position').toBe('below')
```

File: packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx [hunk 7]
```diff
@@ -62,4 +62,24 @@ describe('Toolbar', () => {
...
+          aria-label="Toolbar overflow menu"
+          items={[]}
+          overflow
+          overflowItem={{ title: 'More', menu: { popper: { offset: [0, 4], position: 'below' } } }}
+        />,
+      );
+
+      const overflowItem = wrapper.find('ToolbarItem').last();
+
+      expect(overflowItem.prop('menu')).toMatchObject({
```
Comment (ling1726) [inline packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:79]: nit: do you want to use `toMatchObject` here ? it works off partial matching. So if you want an exact match then `.equals` might be better

Comment (ling1726) [reply packages/fluentui/react-northstar/test/specs/components/Toolbar/Toolbar-test.tsx:79]: @layershifter suggestion looks cleaner, feel free to ignore this

**Review by layershifter — APPROVED** [review]

**Pull request merged**